libraries/AP_NavEKF/EKF_Buffer.cpp

```diff
@@ -10,22 +10,21 @@
 
 // constructor
 ekf_ring_buffer::ekf_ring_buffer(uint8_t _elsize) :
-    elsize(_elsize)
+    elsize(_elsize),
+    buffer(nullptr)
 {}
 
-bool ekf_ring_buffer::init(uint8_t size)
+bool ekf_ring_buffer::init(uint8_t _size)
```

Comment:
Please, please can we have an agreed naming convention! The underscore, not underscore thing is driving me crazy. I personally think it should be underscore for members not underscore for locals, but happy to agree on something different.

Reply:
This seems to confirm to my preference - no-underscores-on-members, underscores-on-locals-if-you-need-them :-)

Reply:
No leading underscores! Okay I realize I'm a minority here, it's okay :)

Reply:
I'm with @andyp1per in that I think arguments shouldn't have underscores.  if there's a name conflict between an argument and a member then let's use something else to differentiate them.  So for example in this case, "size" could be renamed to "buff_size" or "size_max" or something like that.

Underscores on members is ok but optional

Reply:
underscores just make code harder to read and adds absolutely nothing. The header file tells you if its private, protected or public. The _ convention is a bad hangover from C

Reply:
note that _ on constructor and init arguments does help, and is used to init a var.